File: LDC.Atlas.Services.Execution/Entities/InvoiceRecord.cs
using LDC.Atlas.DataAccess.DapperMapper;
using System;
using System.Collections.Generic;

namespace LDC.Atlas.Services.Execution.Entities
{
    public class InvoiceRecord
    {
        public long InvoiceId { get; set; }

        public int DataVersionId { get; set; }

        // DocumentReference
        public string InvoiceLabel { get; set; }

        // ExternalInvoice
        public string ExternalInvoiceRef { get; set; }

        public decimal TotalInvoiceValue { get; set; }

        public InvoiceDocumentType InvoiceDocumentType { get; set; }

        public DateTime DueDate { get; set; }

        public int InvoiceTypeId { get; set; }

        [Column(Name = "Counterparty")]
        public string CounterpartyCode { get; set; }

        public decimal TotalGoodsValue { get; set; }

        public decimal TotalInvoiceQuantity { get; set; }

        public DateTime InvoiceDate { get; set; }

        public string Currency { get; set; }

        public InvoiceType InvoiceType { get; set; }

        public string PaymentTerms { get; set; }

        public QuantityToInvoiceType QuantityToInvoiceType { get; set; }

        public string CompanyId { get; set; }

        public string Template { get; set; }

        public InvoiceSourceType ExternalInhouse { get; set; }

        public DocumentRecord Document { get; set; }

        public TransactionRecord Transaction { get; set; }

        public IEnumerable<InvoiceLineRecord> InvoiceLines { get; set; }

        public long? PhysicalDocumentId { get; set; }

        public long? PreviewPhysicalDocumentId { get; set; }

        public string DocumentReference { get; set; }

        public int YearNumber { get; set; }

        public int TransactionDocumentTypeId { get; set; }

        public bool AuthorizedForPosting { get; set; }

        public long TransactionDocumentId { get; set; }

        public TransactionDocumentType? DocumentType { get; set; }

        public CostDirection? CostDirection { get; set; }

        public string NominalAccountNumber { get; set; }

        public long? BankAccountId { get; set; }

        public DateTime? AgreementDate { get; set; }

        public string C2CCode { get; set; }

        public bool NominalAlternativeAccount { get; set; }

        public bool CostAlternativeCode { get; set; }

        public string DepartmentAlternativeCode { get; set; }

        public bool TaxInterfaceCode { get; set; }

        public bool IsPosted { get; set; }

        public int? PricingOptionId { get; set; }
    }
}

File: LDC.Atlas.WebApp/Client/app/shared/services/execution/dtos/invoice-record.ts
import { DocumentRecord } from '../../../entities/document-record.entity';
import { InvoiceLineRecord } from '../../../entities/invoice-line-record.entity';
import { CostDirections } from '../../../enums/cost-direction.enum';
import { InvoiceTypes } from '../../../enums/invoice-type.enum';
import { QuantityToInvoiceType } from '../../../enums/quantity-to-invoice.enum';
import { TransactionDocumentTypes } from '../../../enums/transaction-document-type.enum';
import { TransactionRecord } from './invoice-transaction-record';

export class InvoiceRecord {
    invoiceId: number;
    invoiceLabel: string;
    externalInvoiceRef: string;
    counterpartyCode: string;
    totalGoodsValue: number; // totalPriceHT: number;
    totalInvoiceValue: number; // totalPriceTTC: number;
    invoiceDate: Date;
    dueDate: Date;
    currency: string;
    invoiceType: InvoiceTypes;
    paymentTerms: string;
    quantityToInvoiceType: QuantityToInvoiceType;
    companyId: string;
    template: string;
    externalInhouse: number;
    document: DocumentRecord;
    transaction: TransactionRecord;
    invoiceLines: InvoiceLineRecord[];
    authorizedForPosting: boolean;
    quantityToInvoice: number;
    transactionDocumentId: number;
    documentType: TransactionDocumentTypes;
    costDirection: CostDirections;
    decimalOption: number;
    invoiceTypeId: number;
    transactionDocumentTypeId: number;
    nominalAccountNumber: string;
    physicalDocumentId?: number;
    isDraft: boolean;
    bankAccountId: number;
    agreementDate: Date;

    c2CCode: string;
    nominalAlternativeAccount: boolean;
    costAlternativeCode: boolean;
    departmentAlternativeCode: string;
    taxInterfaceCode: boolean;

    pricingOptionId: number;
    settlementValue: number;
    documentReference: string;
}
